### PR TITLE
Rewrite torch.ops.aten.select.int to TTNN feasible Aten ops

### PR DESCRIPTION
**Pull request by pschang-phy**
### Ticket

same to this PR https://github.com/tenstorrent/pytorch2.0_ttnn/pull/166, but with different approach.

cc @jdh8 

### Problem description

**torch.ops.aten.select.int** not directly support by TTNN.

### What's changed

I propose add one more stage in `ToTtPass`, the **RewriteAtenOp** stage.

This transformation stage rewrites Aten ops also in the Aten IR. For those **Aten** ops not support directly by or not optimal for TTNN, eg. **torch.ops.aten.select.int**, could rewrite in here. And, let the latter stage to do dedicated lowering into TTNN.

The benefit is that doing ATEN op transformation could only focus on the arithmetic equivalence.

### Test
```bash
$ pytest -s tests/lowering/misc/test_select.py --capture=no
```

### CODE REVIEW

File: tests/lowering/misc/test_select.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    def forward(self, input):
+        return torch.select(input, 3, 0)
+
+
+@pytest.mark.parametrize(
+    "input_shapes",
+    [(1, 32, 16, 3, 96)],
+)
+def test_adaptive_avg_pool_2d(device, input_shapes):
```
Comment (jerrysky3) [inline tests/lowering/misc/test_select.py:21]: Rename to `test_select`?

File: tests/lowering/misc/test_select.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    # The compilation is lazy, so we need to run forward once to trigger the compilation
+    m = torch.compile(m, backend=torch_ttnn.backend, options=option)
+
+    result_after = m.forward(inputs)
+    option._out_fx_graphs[0].print_tabular()
+
+    # Check the graph has be rewritten and contain ttnn ops
+    nodes = list(option._out_fx_graphs[0].nodes)
+    assert [node.target for node in nodes].count(ttnn.slice) == 1
```
Comment (jerrysky3) [inline tests/lowering/misc/test_select.py:37]: Also check if `ttnn.squeeze` exists?

File: tests/lowering/misc/test_select.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+class SelectModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, input):
+        return torch.select(input, 3, 0)
+
+
+@pytest.mark.parametrize(
+    "input_shapes",
```
Comment (jerrysky3) [inline tests/lowering/misc/test_select.py:18]: Maybe we can add select dim and index as test parameters, to enhance the strength of the tests

File: tests/lowering/misc/test_select.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    [(1, 32, 16, 3, 96)],
```
Comment (ayerofieiev-tt) [inline tests/lowering/misc/test_select.py:19]: can we add a couple more inputs shapes please?

File: tests/lowering/misc/test_select.py (hunk 1 above)
Comment (ayerofieiev-tt) [inline tests/lowering/misc/test_select.py:21]: rename method

Comment (pschang-phy) [reply tests/lowering/misc/test_select.py]: done. refer to 5043623e7007c31b066987695337e96cd3878473

Comment (pschang-phy) [reply tests/lowering/misc/test_select.py]: done. refer to 5043623e7007c31b066987695337e96cd3878473

File: tests/lowering/misc/test_select.py (hunk 2 above)
Comment (pschang-phy) [reply tests/lowering/misc/test_select.py]: Refer to 7f025909c0491e369999a3a2778b8ae3577f715a, I check both `ttnn.squeeze` and `ttnn.squeeze`.

`torch.ops.aten.squeeze.dim` maybe lowering to `ttnn.squeeze` or `ttnn.squeeze`.

File: tests/lowering/misc/test_select.py (hunk 3 above)
Comment (pschang-phy) [reply tests/lowering/misc/test_select.py]: done. refer to 08b155c5c66a408a1fe6ffc6109be103747cc444. also add more test parameters.

File: tests/lowering/misc/test_select.py (hunk 4 above)
Comment (pschang-phy) [reply tests/lowering/misc/test_select.py]: Sure, please refer to 08b155c5c66a408a1fe6ffc6109be103747cc444.

File: tests/lowering/misc/test_select.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+from tests.utils import assert_with_pcc
+
+
+class SelectModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, input, dim):
+        return torch.select(input, dim, 0)
```
Comment (jerrysky3) [inline tests/lowering/misc/test_select.py:14]: Maybe also test with different index beyond 0?

File: tests/lowering/misc/test_select.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    # The compilation is lazy, so we need to run forward once to trigger the compilation
+    m = torch.compile(m, backend=torch_ttnn.backend, options=option)
+
+    result_after = m.forward(inputs, dim)
+    option._out_fx_graphs[0].print_tabular()
+
+    # Check the graph has be rewritten and contain ttnn ops
+    nodes = [node.target for node in option._out_fx_graphs[0].nodes]
+    assert nodes.count(ttnn.slice) == 1 and \
+           (nodes.count(ttnn.squeeze) == 1 or nodes.count(ttnn.reshape) == 1)
```
Comment (ayerofieiev-tt) [inline tests/lowering/misc/test_select.py:57]: Can we please instead check that there is no aten.select instead to make this part more robust to any future compiler changes?

**Comment by ayerofieiev-tt** [comment]
@pschang-phy , also make sure to run `pip install pre-commit`, `pre-commit install`

File: tests/lowering/misc/test_select.py (hunk 5 above)
Comment (pschang-phy) [reply tests/lowering/misc/test_select.py]: Sure, refer to 87a3a609ae5e513f380e23eb06a92d943f569c62 adding one more `index` parameter. And more test cases for different index.

File: tests/lowering/misc/test_select.py (hunk 6 above)
Comment (pschang-phy) [reply tests/lowering/misc/test_select.py]: yes, I have changed to check there is no target aten op exists after compiling , refer to https://github.com/tenstorrent/pytorch2.0_ttnn/commit/87a3a609ae5e513f380e23eb06a92d943f569c62

**Review by ayerofieiev-tt — APPROVED** [review]

**Comment by jerrysky3** [comment]
It looks like the merge is blocked by a test already failing on main branch: https://github.com/tenstorrent/pytorch2.0_ttnn/issues/375

**Comment by pschang-phy** [comment]
@ayerofieiev-tt please help to check this CI fail in `tests/lowering/creation/test_masked_fill.py`
https://github.com/tenstorrent/pytorch2.0_ttnn/actions/runs/11789328939/job/32837910200#step:4:605

It went wrong with segmentation fault, but locally is passed
